src/codecs/tiff.rs

```diff
@@ -234,6 +258,16 @@ impl<'a, R: 'a + Read + Seek> ImageDecoder<'a> for TiffDecoder<R> {
             .read_image()
             .map_err(ImageError::from_tiff_decode)?
         {
+            tiff::decoder::DecodingResult::U8(v)
+                if self.original_color_type == ExtendedColorType::Cmyk8 =>
+            {
+                let mut in_cur = Cursor::new(v);
+                let mut out_cur = Cursor::new(buf);
+                let mut cmyk = [0_u8; 4];
+                while in_cur.read_exact(&mut cmyk).is_ok() {
```

Comment:
Could you instead use `chunks_exact` for this iteration?

```rust
for (rgb, cmyk) in buf.chunks_exact(3).zip(v.chunks_exact(4)) {
    rgb.copy_from_slice(&cmyk_to_rgb(cmyk.try_into().unwrap());
}
```

(This would be nicer with array_chunks but that's not stable yet)

Reply:
Thanks, I always forget which ones are stabilized at all.